features/build_id_versioning/only_appropriate_worker_gets_task/feature.go

```diff
@@ -0,0 +1,123 @@
+package only_appropriate_worker_gets_task
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"go.temporal.io/api/common/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/features/features/build_id_versioning"
+	"go.temporal.io/features/harness/go/harness"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/worker"
+	"go.temporal.io/sdk/workflow"
+)
+
+var Feature = harness.Feature{
+	Workflows:     Workflow,
+	Execute:       Execute,
+	WorkerOptions: worker.Options{BuildIDForVersioning: "2.1"},
+}
+
+func Execute(ctx context.Context, r *harness.Runner) (client.WorkflowRun, error) {
+	// Add some versions to the queue
+	err := build_id_versioning.AddSomeVersions(ctx, r.Client, r.TaskQueue)
+	if err != nil {
+		return nil, err
+	}
+
+	// Start workflow
+	run, err := r.ExecuteDefault(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// Wait a few seconds for query to say it's ready for this worker to stop
+	err = r.QueryUntilEventually(ctx, run, "waiting", true, 50*time.Millisecond, 5*time.Second)
```

Comment:
use sync update?

Reply:
Heh, didn't even occur to me. I think I like your other idea more.